docs/README.md

```diff
@@ -1920,60 +1914,277 @@ mutation {
 ```typescript
-await client.subscribe({
-    variables: {
-      productId: <some-product-id>,
-    },
-    query: gql`
-      subscription ProductUpdatesReadModel($productId: ID!) {
-        ProductUpdatesReadModel(id: $productId) {
-          id
-          availability
-          lastUpdate
-          previousUpdate
+import { split, HttpLink } from '@apollo/client';
+import { getMainDefinition } from '@apollo/client/utilities';
+import { WebSocketLink } from '@apollo/client/link/ws';
+import { ApolloClient, InMemoryCache } from '@apollo/client';
+
+function isSubscriptionOperation({ query }) {
+  const definition = getMainDefinition(query);
+  return definition.kind === 'OperationDefinition' && definition.operation === 'subscription'
+}
+
+const httpLink = new HttpLink({
+  uri: '<httpURL>',
+});
+
+// CHANGED: We create an "authLink" that modifies the operation by adding the token to the headers
+const authLink = new ApolloLink((operation, forward) => {
+  operation.setContext({ 
+    headers: { 
+      Authorization: 'Bearer <your token>'
+    }
+  })
+  return forward(operation)
+})
+
+// <-- CHANGED: Concatenate the links so that the "httpLink" receives the operation with the headers set by the "authLink"
+const httpLinkWithAuth = authLink.concat(httpLink) 
+
+const subscriptionClient = new SubscriptionClient('<websocketURL>', {
+  reconnect: true,
+  // CHANGED: added a "connectionParam" property with a function that returns the `Authorizaiton` header containing our token
+  connectionParams: () => {
+    return {
+      Authorization: 'Bearer <your token>'
+    }
+  }
+})
+const wsLink = new WebSocketLink(subscriptionClient);
+
+const splitLink = split(isSubscriptionOperation, wsLink, httpLinkWithAuth); // Note that we now are using "httpLinkWithAuth"
+
+const client = new ApolloClient({
+  link: splitLink,
+  cache: new InMemoryCache()
+});
+```
+
+##### Refreshing tokens with Apollo clients
+
+Authorization tokens expire after a certain amount of time. When a token is expired, you will get an error and you will need to call the [refresh the token](#refresh-token) endpoint to get a new token. After you have done so, you need to use the new token in your GraphQL operations.
+
+There are several ways to do this. Here we show the simplest one for learning purposes. 
+
+First, we modify the example shown in the section ["Sending tokens with apollo clients](#sending-tokens-with-apollo-clients) so that the token is stored in a global variable and the Apollo links get the token from it. That variable will be updated when the user signs-in and the token is refreshed:
+
+```typescript
```

Comment:
The same with this code, I would use 🪓 if it is included in apollo website.

Reply:
This part, especially, is not shown anywhere, and it is hard to figure it out, that's why I wanted to write it here. But! Maybe there are better ways to show it, with less code... not sure. What do you think?

Ideally, this would be shown in an example application, but we still don't have any that uses authentication with tokens as shown here (this is something new)

Reply:
Is Appollo refreshing the token automatically, or is that something the devs should take care of?

If it is not Apollo, the important point is using a global variable, and in that case, I honestly don't like that approach.

I have several concerns about these explanations:
- They are too long.
- We are going deeper into something we should not (front end patterns)
- The global variable approach we are suggesting would make no sense in a reactive client application.
- We are mixing the experience required by the reader trying to teach them how to achieve this, assuming it is difficult for them to figure it out. But at the same time, the snippets have a lot of implicit knowledge, such as the `split` function from Apollo, the `isSubscriptionOperation` from us, the `links`, the `caches`...

I'd start by defining the target audience for this section and their motivations to read this, once clear, talk to them in their idiom. If we target expert readers, we can get rid of the low-level details explanation because they already know. If, on the contrary, we aim less-experienced readers, we should use a language and examples they can understand because they need that explanation.

IMO, this section's readers are here because they want to know how to integrate their clients with a Booster backend. It seems they are front end developers, or maybe backend developers making an integration. In both cases, they are familiar with calling APIs and refreshing tokens.

Maybe, the new thing here is the WebSocket stuff. But teaching that is no the responsibility of the Booster documentation. It is our responsibility to guide the readers, so we make their life's easier, and that includes making great use of their time and curating the links we use.

I totally, get your point in explaining this, I'm not saying we should not do it, we just must sure that we do it in the best possible way.